book-1-orientation/chapters/HTTPCLIENT.md

```diff
@@ -0,0 +1,153 @@
+# HttpClient
+
+You've already experienced making http requests from your browser via `fetch`. While building web applications, we often want to make those requests from our server side code. .NET provides us with the `HttpClient` class that allows us to do that.
+
+We'll first need to install an additional package to make this work. Run the following command in your project (NOTE: This is required only if you are in a Console application. If you are in an ASP.NET project, this is not necessary):
+
+```sh
+dotnet add package Microsoft.AspNet.WebApi.Client
+```
+
+
+## Why Make Requests on the Server?
+In a full stack web application, we know we can make requests to 3rd party APIs from the our client side javascript code. So why would we instead choose to make those calls from our server side code? Some of the benefits of making Http requests server side as opposed to client side are:
+
+- Avoid CORS issues
+- Secure sensitive credentials like keys and passwords by not exposing them in your client side code
+- Users can already have external data available as soon as your page loads
+
+## Making an Http Request
+
+Making a request is fairly simple:
+
+```csharp
+var uri = "https://icanhazdadjoke.com/search?term=cat";
+var client = new HttpClient();
+
+// Set request header to accept JSON
+client.DefaultRequestHeaders
+    .Accept
+    .Add(new MediaTypeWithQualityHeaderValue("application/json"));
+
+var response = await client.GetAsync(uri);
+```
+
+The Joke API will return a JSON response. When this response comes back, .NET will automatically convert it from JSON to a C# object for us. To prepare for this, we need to create a C# class in the shape of the data we expect. Let's look at the raw JSON response that comes back from the API:
+
+```json
+{
+    "current_page": 1,
+    "limit": 20,
+    "next_page": 1,
+    "previous_page": 1,
+    "results": [
+        {
+            "id": "haMJRfF6hFd",
+            "joke": "How do you fix a broken pizza? With tomato paste."
+        },
+        {
+            "id": "xc21Lmbxcib",
+            "joke": "How did the hipster burn the roof of his mouth? He ate the pizza before it was cool."
+        },
+        {
+            "id": "51DAA5Tfaxc",
+            "joke": "What did Romans use to cut pizza before the rolling cutter was invented? Lil Caesars"
+        },
+        {
+            "id": "rc2E6EdiNe",
+            "joke": "Want to hear my pizza joke? Never mind, it's too cheesy."
+        }
+    ],
+    "search_term": "pizza",
+    "status": 200,
+    "total_jokes": 4,
+    "total_pages": 1
+}
+```
+
+Given the shape of this data, create a new C# class (or classes) that matches these properties.
+
+```csharp
+public class JokeReponse
+{
+    [JsonProperty("current_page")]
```

Comment:
This is assuming newtonsoft, right? IIRC that’s not installed by default in asp.net core 3.0. MS is trying to use their own json serialization bits. 

I think it’s ok to use newtonsoft, but it should be mentioned.